src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractSyncNamespaceService.State.cs

```diff
@@ -0,0 +1,297 @@
+                if (documents.Any(doc =>
+                {
+                    var defaultNS = doc.Project.DefaultNamespace;
+                    defaultNamespaceFromProjects.Add(defaultNS);
+                    return defaultNS == null;
+                }))
+                {
+                    return (false, null, null);
+                }
+
+                if (defaultNamespaceFromProjects.Count == 1)
+                {
+                    var defaultNamespace = defaultNamespaceFromProjects.Single();
+
+                    // If the cursor location doesn't meet the requirement to trigger the refactoring in any of the documents 
+                    // (See `ShouldPositionTriggerRefactoringAsync`), or we are getting different namespace declarations among 
+                    // those documents, then we know we can't make a proper code change. We will return false and the refactoring 
+                    // will then bail. We use span of namespace declaration found in each document to decide if they are identical.
+                    var spansForNamespaceDeclaration = new Dictionary<TextSpan, TNamespaceDeclarationSyntax>();
+                    foreach (var document in documents)
+                    {
+                        (var shouldTrigger, var namespaceDeclaration) =
+                        await service.ShouldPositionTriggerRefactoringAsync(document, textSpan.Start, cancellationToken)
+                        .ConfigureAwait(false);
+
+                        // In case there's no namespace declaration in the document, we used an empty span as key, 
+                        // since a valid namespace declaration node can't have zero length.
+                        var span = namespaceDeclaration?.Span ?? new TextSpan(0, 0);
+                        if (shouldTrigger)
+                        {
+                            spansForNamespaceDeclaration[span] = namespaceDeclaration;
+                        }
+                        else
+                        {
+                            return (false, null, null);
+                        }
+                    }
+
+                    if (spansForNamespaceDeclaration.Count == 1)
+                    {
+                        var namespaceDecl = spansForNamespaceDeclaration.Values.Single();
+                        var declaredNamespace = namespaceDecl == null
+                            // namespaceDecl == null means the target namespace is global namespace.
+                            ? string.Empty
+                            // Since the node in each document has identical type and span, 
+                            // they should have same name.
+                            : SyntaxGenerator.GetGenerator(documents.First()).GetName(namespaceDecl);
+
+                        return (true, defaultNamespace, declaredNamespace);
+                    }
+                }
+
+                return (false, null, null);
```

Comment:
either only have a single bail-out path.  or test and bail early.  mixing both styles is very confusing.